Monika-After-Story/MonikaModDev
# Safe EV modification functions

Alternate solution to #3949. Upon thinking about it, a dummy ev isn't necessarily the greatest solution in some circumstances.

This PR modifies a bunch of `mas_getEV` calls to use a safe variant instead, so they're all naturally error handled.

### This PR adds the following functions:
#### `mas_checkEV(ev_label, predicate)`:
- Checks event properties given the event's eventlabel, using a lambda predicate function. If the event doesn't exist, `False` is returned

#### `mas_getEVPropValue(ev_label, prop, default=None)`:
- Gets a specified property from the event corresponding to the given eventlabel. If the property or event doesn't exist, the default value is returned

#### `mas_setEVPropValues(ev_label, **kwargs)`:
- Sets the properties of an event given its eventlabel using kwargs (arg names match property names)

#### `mas_isPoolEV(ev_label)`:
- Checks if the given event for the eventlabel is pooled

#### `mas_isRandomEV(ev_label)`:
- Checks if the given event for the eventlabel is random

#### `mas_isUnlockedEV(ev_label)`:
- Checks if the given event for the eventlabel is unlocked

#### `mas_getEV_last_seen(ev_label, default=None)`:
- Gets the event's `last_seen` if it exists, otherwise returns the default

#### `mas_getEV_shown_count(ev_label, default=0)`:
- Gets the event's `shown_count` if it exists, otherwise returns the default

#### `mas_inRulesEV(ev_label, *args)`:
- Checks if all given rule keys (provided `*args`) are present in the event's rules dict

#### `mas_addignModifyEVPropValue(ev_label, propname, operation, value)`:
- Performs an assign-modify operation on an event's prop value, generalized such that it works with all prop values and assign-modify operations

***These are essentially safe event object getters/setters/checkers)***


## Testing:
- Please verify the new functions work properly and safely.
- Verify no crashes
- Verify topics still work as normal
- Verify ev data is set correctly
